internal/corazawaf/transaction.go

```diff
@@ -1408,6 +1408,11 @@ func (tx *Transaction) AuditLog() *auditlog.Log {
 					})
 				}
 			}
+		case types.AuditLogPartFinalBoundary:
+			FinalBoundary = true
+		}
+		if FinalBoundary {
```

Comment:
Hmm - I know @jcchavezs won't like it since it's an API change ;) But if we can still do it, we should remove `types.AuditLogPartFinalBoundary`, this thing really only makes sense when parsing strings, not as a real piece of code, and is probably causing some nanoseconds of extra work with absolutely no purpose. @jcchavezs what do you think?

Regardless of whether we remove the above or not, it probably makes sense to check `Z` when parsing and make it an error if any characters are specified after it.

Reply:
I'm afraid few parsers use it. Once I made an auditlog parser which uses it (but I do not use it at all).

Reply:
But does it make any difference in the output? Will it print something @airween ?

Reply:
@airween @dune73 can A ever be not at the beginning and Z be not at the end?

Reply:
BTW, my suggestion is about removing the Go code `types.AuditLogPartFinalBoundary` because `Z` is a no-op, hence it's job is to be ignored and doesn't need to be modeled. It's not about removing support for the character itself which I guess many configs probably use. Deprecating it could be fine but the suggestion was specifically about the Go code.

Reply:
I think the Z is there to mark the end in a format that can have optional parts enabled. If you write in a format where the end is defined (-> terminating `}` as in JSON perhaps), then you probably do not need the Z.

Also: With ModSec there is a strange behaviour with interleaving audit log entries that I have not seen in years. The Z helps in this context too, since it clearly tells you that record X is now done.

Reply:
More to the point: No, I do not think you need to expect a different order. All I ever saw was A to Z.